iothub/device/src/Transport/HttpTransportHandler.cs

```diff
@@ -180,8 +185,8 @@ internal async Task UploadToBlobAsync(string blobName, Stream source, Cancellati
             try
             {
                 // 2. Use SAS URI to send data to Azure Storage Blob (PUT)
-                CloudBlockBlob blob = new CloudBlockBlob(new Uri(putString));
-                var uploadTask = blob.UploadFromStreamAsync(source);
+                var blob = new CloudBlockBlob(new Uri(putString));
+                Task uploadTask = blob.UploadFromStreamAsync(source);
                 await uploadTask.ConfigureAwait(false);
```

Comment:
Any reason why the former statement is not just awaited, rather than being split up like this?

Reply:
I actually changed it initially, there is some logic below that I didn't change that relies on the task variable.
I can look into it and see if there is any value in having them around.